lib/app.rb

```diff
@@ -1,4 +1,6 @@
 class App
+  attr_reader :current_project, :input_stream, :last_input, :output_stream, :projects
```

Comment:
Listing these out like this helped me understand what all this class is responsible for. The sea of instance variables in the code made those responsibilities murkier.

And seeing them listed it's clear that class does too much, but I don't feel confident extracting stuff based on the one bug fix so far, because I'm scared of going down the wrong path.